utils/S3UtilsMongoClient.js

```diff
@@ -74,6 +74,92 @@ class S3UtilsMongoClient extends MongoClientInterface {
         }
     }
 
+    async updateInflightDeltas(allMetrics, log) {
+        let cursor;
+        try {
+            const bucketKeys = [];
+
+            if (!allMetrics || !Array.isArray(allMetrics) || allMetrics.length === 0) {
+                return allMetrics;
+            }
+
+            const usersBucketCreationDatesMap = await this._getUsersBucketCreationDates(log);
+            Object.keys(usersBucketCreationDatesMap).forEach(bucketEntry => {
+                const bucketName = bucketEntry.split(constants.splitter)[1];
+                const bucketKey = `bucket_${bucketName}_${new Date(usersBucketCreationDatesMap[bucketEntry]).getTime()}`;
+                bucketKeys.push(bucketKey);
+            });
+
+            cursor = await this.getCollection(INFOSTORE).find({
+                _id: { $in: bucketKeys },
+            }, {
+                projection: {
+                    'usedCapacity._inflight': 1,
+                },
+            });
+
+            const inflights = await cursor.toArray();
+
+            const accountInflights = {};
+            allMetrics.forEach(entry => {
+                const id = entry._id;
+                if (id.startsWith('bucket_')) {
+                    const inflightDocument = inflights.find(i => i._id === id);
+                    const inflight = inflightDocument && inflightDocument.usedCapacity ? inflightDocument.usedCapacity._inflight : 0;
```

Comment:
should we not substract the `preScan` inflights?

```suggestion
                    const inflight = inflightDocument && inflightDocument.usedCapacity ? max(inflightDocument.usedCapacity._inflight - entry.usedCapacity._inflightsPreScan, 0): 0;
```

Reply:
yes, I removed it by mistake when redoing the logic...